### ISSUE TITLE
Pragmas on continued long lines don't work

### ISSUE THREAD

**Issue by nedbat** [coveragepy/coveragepy#754]
(From https://stackoverflow.com/questions/54111889/coverage-pragma-no-branch-with-multiline-statement)

A pragma comment on the continuation lines of a multi-line if-statement won't have an effect:

For the python coverage package, a missing else can be ignored for the resulting coverage by using # pragma: no branch.

It seems, that this is not working when breaking a long if statement in multiple lines like this:
```
if this_is_a_verylong_boolean_expression == True and another_long_expression \
    and here_another_expression:  # pragma: no branch
    do_something()
```

**Comment by devdanzin** [comment]
This happens because `coverage.results.Analysis.arcs_missing()` depends on `coverage.parser.PythonParser.lines_matching()` to collect lines marked as `pragma: no branch`, but `lines_matching()` records the raw line number of the match, not taking multi-line statements into account.

If we record the lines by mapping them using `parser._multiline` instead, the pragma works.

**Comment by nedbat** [comment]
Fixed in commit 277c8c43

**Comment by nedbat** [comment]
This is now released as part of [coverage 7.5.1](https://pypi.org/project/coverage/7.5.1).
